conda_build/__init__.py

```diff
@@ -21,3 +23,13 @@
     'sign',
     'skeleton',
 ]
+
+# Set default logging handler to avoid "No handler found" warnings.
+try:  # Python 2.7+
+    from logging import NullHandler
+except ImportError:
+    class NullHandler(logging.Handler):
+        def emit(self, record):
+            pass
+
+logging.getLogger(__name__).addHandler(NullHandler())
```

Comment:
Maybe you're not done with this PR yet, so this could be a premature comment.
Using NullHandler is exactly the right thing to do for a library.  It makes your library a good logging citizen.  Also, the package's root-level `__init__.py` file is almost always the right place to invoke the basic logging configuration.  (In conda's case I'm not, because I don't want to bog down the `..activate` etc cli stuff with logger initialization.  It's way too slow already.)

However, unless you add another handler somewhere else, the problem with conda-build logs will remain.  I'm guessing you're just not done with the PR though.